PowerUp/app/src/main/java/powerup/systers/com/AvatarRoomActivity.java

```diff
@@ -65,13 +65,33 @@ public void onCreate(Bundle savedInstanceState) {
             SessionHistory.hasPreviouslyCustomized = true;
         } else {
             // if yes, check the customization from database for eyes, skin, hair, skin
-            eye=getmDbHandler().getAvatarEye();
-            skin=getmDbHandler().getAvatarSkin();
-            hair=getmDbHandler().getAvatarHair();
-            cloth=getmDbHandler().getAvatarCloth();
+            source.getAvatarEye(new IDataSource.LoadIntegerCallback() {
+                @Override
+                public void onResultLoaded(int value) {
+                    eye = value;
+                }
+            });
+            source.getAvatarSkin(new IDataSource.LoadIntegerCallback() {
+                @Override
+                public void onResultLoaded(int value) {
+                    skin = value;
+                }
+            });
+            source.getAvatarHair(new IDataSource.LoadIntegerCallback() {
+                @Override
+                public void onResultLoaded(int value) {
+                    hair = value;
+                }
+            });
+            source.getAvatarCloth(new IDataSource.LoadIntegerCallback() {
+                @Override
+                public void onResultLoaded(int value) {
+                    cloth = value;
+                }
+            });
 
             String eyeImageName = getResources().getString(R.string.eye);
-            eyeImageName = eyeImageName + getmDbHandler().getAvatarEye();
+            eyeImageName = eyeImageName + eye;
```

Comment:
I would recommend to check the value, before assignment as its coming from callback.